### PR TITLE
Added records article

### PR DESCRIPTION
**Pull request by cayhorstmann**

### CODE REVIEW

File: site/content/features/records.md [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+```
+record SleepyPoint(double x, double y) {
+   public SleepyPoint throws InterruptedException { // Error
+     Thread.sleep(1000); 
+   }
+}
+```
+
+Reflection
+==========
```
Comment (marchof) [inline site/content/features/records.md]: This should be probably h2 not h1? So 

    Reflection
    ----------

File: site/content/features/records.md [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+   case instanceof Point(x, 0) p: ... // Maybe the future—not in JDK 14
+   ...
+}
+```
+
+
+
+## References
+
+* [JEP 395: Records (Preview), OpenJDK](http://openjdk.java.net/jeps/395)
```
Comment (marchof) [inline site/content/features/records.md]: Link text and link itself should probably be `359`

**Comment by marchof** [comment]
@cayhorstmann  I have a formal remark: I stumbled upon the dashes in the first paragraph without spaces. The font we use renders the dashes very close to the words which at least for my eyes makes it hard to read. What is your take on spaces around dashes?

File: site/content/features/records.md [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+   public SleepyPoint throws InterruptedException { // Error
+     Thread.sleep(1000); 
+   }
+}
+```
+
+Reflection
+==========
+
+The `isRecord` method can tell whether a `Class` instance is a record:
```
Comment (marchof) [inline site/content/features/records.md]: Period instead of colon?

File: site/content/features/records.md [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+   }
+}
+{{< /sandboxsource >}}
+{{< /sandbox >}}
+
+Because arrays are mutable, you can change the elements of `coordinates`.
+
+This is not a good idea, but the Java language won't stop you. In general, Java has no mechanism for expressing immutability. It is no different with reccords.
+
+4\. The implementations of `hashCode`, `equals`, and `toString` in the JDK are _not normative_. In particular, the current behavior of combining two hash codes as `31 * h1 + h2` could change. The behavior of `equals` is constrained by the general `Object.equals` contract, but there is no guarantee that the order of comparisons is fixed. You should not rely on the exact format of the`toString` result either.
```
Comment (marchof) [inline site/content/features/records.md]: Missing blank between "the" and `toString`

**Comment by marchof** [comment]
@cayhorstmann Great content! I did a review and found some formal issues that may be worth fixing before publishing it.

File: site/content/features/records.md (hunk 1 above)
Comment (cayhorstmann) [reply site/content/features/records.md]: Yes

File: site/content/features/records.md (hunk 2 above)
Comment (cayhorstmann) [reply site/content/features/records.md]: Yes

File: site/content/features/records.md (hunk 3 above)
Comment (cayhorstmann) [reply site/content/features/records.md]: Yes

**Comment by cayhorstmann** [comment]
> @cayhorstmann I have a formal remark: I stumbled upon the dashes in the first paragraph without spaces. The font we use renders the dashes very close to the words which at least for my eyes makes it hard to read. What is your take on spaces around dashes?

There is no consensus in English writing whether or not to put spaces around an em-dash. I have never done it myself, but if you prefer them, I can adapt.

**Pull request merged**